src/query/stream/streams.cpp

```diff
@@ -232,14 +236,23 @@ Streams::StreamsMap::iterator Streams::CreateConsumer(StreamsMap &map, const std
         spdlog::trace("Commit transaction in stream '{}'", stream_name);
         interpreter->CommitTransaction();
         result.rows.clear();
-      };
-
-  auto insert_result = map.try_emplace(
-      stream_name, StreamData<TStream>{std::move(stream_info.common_info.transformation_name), std::move(owner),
-                                       std::make_unique<SynchronizedStreamSource<TStream>>(
-                                           stream_name, std::move(stream_info), std::move(consumer_function))});
-  MG_ASSERT(insert_result.second, "Unexpected error during storing consumer '{}'", stream_name);
-  return insert_result.first;
+      } catch (const query::TransactionSerializationException &e) {
+        if (i == total_retries) {
+          throw;
+        }
+        using namespace std::chrono_literals;
+        std::this_thread::sleep_for(500ms);
+      }
+      break;
+    }
+
+    auto insert_result = map.try_emplace(
+        stream_name, StreamData<TStream>{std::move(stream_info.common_info.transformation_name), std::move(owner),
+                                         std::make_unique<SynchronizedStreamSource<TStream>>(
+                                             stream_name, std::move(stream_info), std::move(consumer_function))});
+    MG_ASSERT(insert_result.second, "Unexpected error during storing consumer '{}'", stream_name);
+    return insert_result.first;
+  };
```

Comment:
```suggestion
   };
    auto insert_result = map.try_emplace(
        stream_name, StreamData<TStream>{std::move(stream_info.common_info.transformation_name), std::move(owner),
                                         std::make_unique<SynchronizedStreamSource<TStream>>(
                                             stream_name, std::move(stream_info), std::move(consumer_function))});
    MG_ASSERT(insert_result.second, "Unexpected error during storing consumer '{}'", stream_name);
    return insert_result.first;
```
I think something like this should fix errors.